pkg/healthcheck/checkers.go

```diff
@@ -76,6 +79,45 @@ func CheckDirectoryExists(path string) Checker {
 	}
 }
 
+// CheckCommandStatus returns a checker which executes a command and returns successfully or
+// unsuccessfully depending on the exit status of the command.
+func CheckCommandStatus(ctx context.Context, cmd string, args ...string) Checker {
+	return func() (bool, string, error) {
+		cmd := exec.CommandContext(ctx, cmd, args...)
+		err := cmd.Start()
+		if err != nil {
+			// for example, command does not exist. This is an error with the check.
+			return false, fmt.Sprintf("failed to start command. `%s` (%v)", cmd, err), err
+		}
+		err = cmd.Wait()
+		if err != nil {
+			// command returns with a non-zero exit code. Return nil error, and false for the check.
+			return false, fmt.Sprintf("command failed `%s` (%v)", cmd, err), nil
+		}
+		// command returns with a zero exit code.
+		return true, fmt.Sprintf("command completed successfully `%s`", cmd), nil
+	}
+}
+
+// CheckK8sPods returns a checker which verifies the number of pods found matches the number
+// expected. If Listing the pods returns an error, the error is returned. The boolean value returned
+// by the check follows whether the number of pods observed in the list matches the expected count.
```

Comment:
This method intentionally doesn't check the Status of the pods to verify they are all running, only that they exist in the correct number.

My rational for this is that kubernetes is generally self-correcting, so the fix action for a pod not being started is to do nothing and the scheduler will eventually succeed. Unlike local runners, testground is not the scheduler/supervisor for pods on k8s, and can be of no assistance when it comes to restarting pods once they are in the control of the scheduler.

As it is, the Fixer which would fix the problem detected by CheckK8sPods  (not yet written) would apply a new pod configuration through the API, or using kubectl apply, or helm install, etc.

On the other hand, perhaps it would be worthwhile to know before starting that the right number of pods exist. I could imagine a Fixer which watched/waited for pods to become ready. Waiting on pods to become ready is a different category of problem, and I want to distinguish between them.